pdc/apps/compose/views.py

```diff
@@ -1276,6 +1277,110 @@ def _clear(self, release_obj, args):
+
+        And if overrides-rpm have exited in target-release, they don't get copied.
+
+        __Method__: POST
+
+        __URL__: $LINK:overridesrpmclone-list$
+
+        __DATA__:
+
+            {
+                "source_release_id":        string
+                "target_release_id":        string
+                "rpm_name":                 string      #optional
+                "srpm_name":                string      #optional
+                "variant":                  string      #optional
+                "arch":                     string      #optional
+            }
+
+        __Response__:
+
+            [
+                {
+                    "arch":                                     "string",
+                    "comment (optional, default=\"\")":         "string",
+                    "do_not_delete (optional, default=false)":  "boolean",
+                    "id (read-only)":                           "int",
+                    "include":                                  "boolean",
+                    "release":                                  "Release.release_id",
+                    "rpm_arch":                                 "string",
+                    "rpm_name":                                 "string",
+                    "srpm_name":                                "string",
+                    "variant":                                  "string"
+                }
+            ]
+        """
+        data = request.data
+        keys = set(['source_release_id', 'target_release_id'])
+        arg_filter_map = ['variant', 'arch', 'srpm_name', 'rpm_name', 'rpm_arch']
+        allowed_keys = list(keys) + arg_filter_map
+        extra_keys = set(data.keys()) - set(allowed_keys)
+        if extra_keys:
+            return Response({'detail': '%s keys are not allowed' % list(extra_keys)},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+        if 'source_release_id' not in data:
+            return Response({'detail': 'Missing source_release_id'},
+                            status=status.HTTP_400_BAD_REQUEST)
+        source_release_id = data.pop('source_release_id')
+        if 'target_release_id' not in data:
+            return Response({'detail': 'Missing target_release_id'},
+                            status=status.HTTP_400_BAD_REQUEST)
+        target_release_id = data.pop('target_release_id')
+        try:
+            source_release = get_object_or_404(Release, release_id=source_release_id)
+        except Http404:
+            return Response({'detail': 'Source_release %s is not existed' % source_release_id},
+                            status=status.HTTP_404_NOT_FOUND)
+        try:
+            target_release = get_object_or_404(Release, release_id=target_release_id)
+        except Http404:
+            return Response({'detail': 'Target_release %s is not existed' % target_release_id},
+                            status=status.HTTP_404_NOT_FOUND)
+
```

Comment:
I agree with @ycheng-aa that if this logic could be abstracted into a separate function it would be better.

Anyway, using `get_object_or_404` does not have any advantages here. Why not use `Release.objects.get()` directly and catch `Release.DoesNotExist`?

Grammar note: the error should be `Target release %s does not exist.`.